File: src/main/java/org/amithalpert/GameScreen.java
package org.amithalpert;
import java.awt.*;
import javax.swing.*;

public class GameScreen extends JPanel
{
	Player player;
	FlappyNix nix;
	Image worldTexture;

	
	public GameScreen()
	{
		worldTexture = (new ImageIcon("src/main/resources/background.png")).getImage();
		player = new Player(this);
		nix = new FlappyNix(player, this);


		addKeyListener(player);
		setFocusable(true);
	}
	
	public void paintComponent(Graphics g)	
	{
		super.paintComponent(g);
		
		g.drawImage(worldTexture,0,0,getWidth(),getHeight(),null);

		g.setColor(Color.red);
		g.drawRect(nix.groundCollider.x , nix.groundCollider.y , nix.groundCollider.width, nix.groundCollider.height);
		g.drawRect(player.hitBox.x, player.hitBox.y , player.hitBox.width, player.hitBox.height);
		

		player.draw(g);
	}



	public static void main(String[] args) 
	{
		Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();

		JFrame frame = new JFrame("FlappyNix");
		GameScreen gameScreen = new GameScreen();
		frame.add(gameScreen);
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		frame.setSize(screenSize.width, screenSize.height);
		frame.setResizable(false);
		frame.setVisible(true);
		frame.setFocusable(false);

	}
}

File: src/main/kotlin/GameScreen.kt
import java.awt.*
import java.awt.event.KeyEvent
import java.awt.event.KeyListener
import javax.swing.ImageIcon
import javax.swing.JFrame
import javax.swing.JPanel

class GameScreen : JPanel() {


    var player: Player
    var background: Image? = null
    var moveFlag = false

    init {
        background = ImageIcon("src/main/resources/background.png").image
        player = Player(this)

        addKeyListener(player)

        isFocusable = true
    }


    public override fun paintComponent(g: Graphics) {
        super.paintComponent(g)
        g.drawImage(background, 0, 0, width, height, null)

        player.draw(g)

    }


    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            // Get screen size
            val screenSize = Toolkit.getDefaultToolkit().screenSize
            val ScreenWidth = screenSize.getWidth()
            val ScreenHeight = screenSize.getHeight()

            // setup Jframe
            val frame = JFrame("FlappyNix")
            val bp = GameScreen()

            frame.add(bp)

            frame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
            frame.setSize(ScreenWidth.toInt(), ScreenHeight.toInt())
            frame.isResizable = false
            frame.isVisible = true
            frame.isFocusable = false
            frame.revalidate();
            frame.repaint();


            try {
                Thread.sleep(50)
            } catch (e: InterruptedException) {
            }
            bp.moveFlag = true
        }
    }
}